search_params.py

```diff
@@ -94,7 +95,13 @@ def init_search_params_spaces(model_config):
                      'qloguniform', 'randn', 'qrandn', 'randint', 'qrandint']
     for key, value in model_config.items():
         if isinstance(value, list) and len(value) >= 2 and value[0] in search_spaces:
-            model_config[key] = getattr(tune, value[0])(*value[1:])
+            search_space, search_args = value[0], value[1:]
+            # If the values in search values are lists (e.g., [2] in [[2], [4], [6]]), the search space must be `grid_search`.
+            if len(search_args) > 0 and any(isinstance(x, list) for x in search_args) and search_space != 'grid_search':
+                raise ValueError(
+                    'Only `grid_search` can assign list of search spaces.')
```

Comment:
I feel a sentence like the one in line 99 (after revising the example) is clearer